### DISCUSSION TITLE
`tea -E` analog for use in `direnv`?

### DISCUSSION THREAD

**Discussion by jasisk** [pkgxdev/pkgx#779] [Help]
Hey there.

Was curious if there was a kosher way to replicate the behavior of `tea -E` in this new v1.0.0 world. Specifically, I was using the following [direnv](https://direnv.net/) plugin to allow for dev env magic without magic:

```sh
use_tea() {
	direnv_load tea -E -- direnv dump
}

watch_file .tea.yml
```

Would something like this be legit and something not prone to breaking in the future?

```sh
use_pkgx() {
	direnv_load pkgx --internal.activate "${PWD}" -- direnv dump
}

watch_file .pkgx.yaml
```

**Comment by mxcl** [comment]
This may break because yeah it's internal, but we can make a more formal way to do this which we then promise to keep.

What does `direnv dump` do?

**Reply by jasisk** [reply]
Felt gnarly asking about using an internal flag so the potential for breakage is totally expected—frankly, just hoped it would help convey the question better.

Adding a formal method would be greatly appreciated—thank you!

`direnv dump` just (safely) serializes the env in a format that can be loaded by `direnv_load`.

Basically, this pseudocode:

```
const vars = getEnvVars();
const escapedVars = escapeKeysAndValues(vars);

return base64Enc(gzip(JSON.stringify(escapedVars)));
```

**Reply by mxcl** [reply]
This seems like a neat thing to add to our docs, once we formalize the args to `dev`. Let me think about how to approach this and get back to you.

**Comment by jasisk** [comment]
If folks wander in here looking for a solution while the api is considered, the following works just fine. `direnv` will just pass along the exported environment variables, not the functions or hooks. This does not use the `direnv` escaping bits provided by `direnv_load` and `direnv dump` but `pkgx` escapes the vars it exports and we're not reexporting anything else.

> [!WARNING]
> **HUGE CAVEAT!**
> This is using an internal flag that has no guarantees of stability. This `direnv` plugin may break at any time and it's not `pkgx` / `pkgx` maintainer's fault.
>
> As a result, I won't be marking this as an answer. Buyer beware. 😃

```shell
use_pkgx() {
	# THIS USES AN INTERNAL AND UNSUPPORTED FLAG
	eval "$(pkgx --internal.activate "${PWD}")"
}

watch_file pkgx.yaml .pkgx.yaml
``` 

usage in an `.envrc` file:
```
use pkgx
```

**Comment by adam50200** [comment]
No analogy
